### DISCUSSION TITLE
DOCKER_BUILDKIT=1 requirement

### DISCUSSION THREAD

**Discussion by jsonn** [docker-mailserver/docker-mailserver#3749] [General]
While trying to run the test system, I stumbled over the need for `DOCKER_BUILDKIT=1` on both Debian Bookworm and Sid. Should this be set by default by make or do I miss something else?

**Comment by polarathene** [comment]
- [Docker image building minimum requirements](https://docker-mailserver.github.io/docker-mailserver/latest/examples/tutorials/docker-build/#minimum-supported-version) (_[Recent change](https://github.com/docker-mailserver/docker-mailserver/pull/3666/files#diff-0b0bce140cb465cd0540de837bef2da637e0a037e37aa65cd162e096132296f4) dropped `DOCKER_BUILDKIT=1` from the `Makefile`_):
  > _We make use of build features that require a recent version of Docker. v23.0 or newer is advised, but earlier releases may work._
  > _If you are using a version of Docker prior to v23.0, you will need to enable BuildKit via the ENV [DOCKER_BUILDKIT=1](https://docs.docker.com/build/buildkit/#getting-started)_
- [Related](https://github.com/docker-mailserver/docker-mailserver/pull/3729#issuecomment-1870873840)

---

> Should this be set by default by make or do I miss something else?

You should install Docker from official upstream (_as per Dockers docs for your distro_).

**Reply by jsonn** [reply]
Well, `apt install docker-compose` is the official Debian way for doing it and Debian is explicitly recommended in the documentation. Which makes it overall a bad UX.

**Reply by polarathene** [reply]
> Which makes it overall a bad UX.

We're aligned with official Docker docs which is compose v2. So you should be using `docker compose` commands, not `docker-compose`.

We explicitly state in our docs to build the image requires v23 or newer (Feb 2023), otherwise you can try adding the ENV `DOCKER_BUILDKIT=1`.

---

> Well, apt install docker-compose is the official Debian way for doing it and Debian is explicitly recommended in the documentation.

Tests run without compose currently, when compose is adopted it'll be expected to be v2 compatible only.

https://docs.docker.com/engine/install/debian/#installation-methods

> Set up and install Docker Engine from [Docker's `apt` repository](https://docs.docker.com/engine/install/debian/#install-using-the-repository).

https://docs.docker.com/engine/install/debian/#install-using-the-repository

> Before you install Docker Engine for the first time on a new host machine, you need to set up the Docker `apt` repository. Afterward, you can install and update Docker from the repository.

https://docs.docker.com/engine/install/debian/#install-from-a-package

> If you can't use Docker's `apt` repository to install Docker Engine, you can download the `deb` file for your release and install it manually. You need to download a new file each time you want to upgrade Docker Engine.

[Compose docs also suggest the same](https://docs.docker.com/compose/install/linux/#install-using-the-repository), you perhaps misread them? I don't see any explicit _"use your distro package"_ advice in these docs, only to add the Docker repo to your package manager and install the package from that.
